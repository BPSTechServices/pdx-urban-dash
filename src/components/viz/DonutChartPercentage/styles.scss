/*
These are the global variables we use. Just an optimization to keep
us using standardized widths and colors. Reference base-styles.scss when
defining font-families, sizes, margins, padding or colors.
*/
@import "../../../base-styles";

.DonutChartPercentage {

  display: block;
  margin: auto;

  &-wrapper {
    height: 100%;
    width: 100%;
  }

  &-title {
    font-family: $font-heading;
    font-size: $font-size-lg;
    margin-left: auto;
    margin-right: auto;
  }

  &-theme {
    &__light {
      color: $color-font-dark;
      background-color: white;
    }

    &__dark {
      color: $color-font-light;
      background-color: $color-font-dark;
    }
  }

  &-track {
    fill: transparent;
    stroke: #DAE2E5;
  }

  &-indicator {
    fill: transparent;
    stroke: #0F0F0F;
  }
  
  &-text-label {
    font-size: $font-size-xsm;
    font-style: $font-heading;
  }
}