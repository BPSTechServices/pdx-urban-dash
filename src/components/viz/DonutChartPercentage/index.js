import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

import Legend from '../shared/Legend';
import ChartWrapper from '../shared/ChartWrapper';

const DonutChartPercentage = (
  {
    title,
    className,
    label,
    color,
    description,
    target,
    value,
  },
) => {
  const percentage = Math.round((value / target) * 100);
  const size = 200;
  const strokeWidth = 50;

  const halfsize = 100;
  const radius = halfsize - 25;
  const circumference = 2 * Math.PI * radius;
  const strokeval = (((percentage) * circumference) / 100);
  const dashval = (`${strokeval} ${circumference}`);

  const trackstyle = { strokeWidth };
  const indicatorstyle = { strokeWidth, strokeDasharray: dashval };
  const rotateval = `rotate(-90 ${halfsize},${halfsize})`;
  return (
    <ChartWrapper title={title} className={className}>
      <svg width={size} height={size} className="DonutChartPercentage">
        <circle r={radius} cx={halfsize} cy={halfsize} transform={rotateval} style={trackstyle} className="DonutChartPercentage-track" />
        <circle r={radius} cx={halfsize} cy={halfsize} transform={rotateval} style={indicatorstyle} className="DonutChartPercentage-indicator" />
        <text className="DonutChartPercentage-text" x={halfsize} y={halfsize} style={{ textAnchor: 'middle' }}>
          <tspan className="DonutChartPercentage-text-val">{percentage}</tspan>
          <tspan>%</tspan>
          <tspan className="DonutChartPercentage-text-label" x={halfsize} y={halfsize + 10}>{label}</tspan>
        </text>
      </svg>
      <Legend
        series={[{ title, description, color }]}
      />
    </ChartWrapper>
  );
};

DonutChartPercentage.propTypes = {
  target: PropTypes.number.isRequired,
  value: PropTypes.number.isRequired,
  title: PropTypes.string,
  className: PropTypes.string,
  color: PropTypes.string,
  description: PropTypes.string,
  label: PropTypes.string,
};

DonutChartPercentage.defaultProps = {
  title: 'chart title',
  className: '',
  color: 'black',
  description: '',
  label: 'Completed',
};

export default DonutChartPercentage;
