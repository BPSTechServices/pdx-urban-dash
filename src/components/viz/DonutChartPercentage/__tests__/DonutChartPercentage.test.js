import React from 'react';
import { shallow } from 'enzyme';
import DonutChartPercentage from '../index';


describe('<LineChart />', () => {
  test('snapshot', () => {
    const wrapper = shallow(
      <DonutChartPercentage
        title="percentage chart"
        className=""
        color="black"
        description="this chart shows a percentage of a value to a target"
        target={75}
        value={40}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
