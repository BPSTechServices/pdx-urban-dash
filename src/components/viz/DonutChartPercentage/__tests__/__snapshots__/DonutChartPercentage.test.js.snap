// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD
exports[`<DonutChartPercentage /> snapshot 1`] = `
=======
exports[`<LineChart /> snapshot 1`] = `
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DonutChartPercentage
    className=""
    color="black"
    description="this chart shows a percentage of a value to a target"
    label="Completed"
    target={75}
    title="percentage chart"
    value={40}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <svg
<<<<<<< HEAD
          className="donutchartpercentage"
=======
          className="DonutChartPercentage"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
          height={200}
          width={200}
        >
          <circle
<<<<<<< HEAD
            className="donutchartpercentage-track"
=======
            className="DonutChartPercentage-track"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
            cx={100}
            cy={100}
            r={75}
            style={
              Object {
                "strokeWidth": 50,
              }
            }
            transform="rotate(-90 100,100)"
          />
          <circle
<<<<<<< HEAD
            className="donutchartpercentage-indicator"
=======
            className="DonutChartPercentage-indicator"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
            cx={100}
            cy={100}
            r={75}
            style={
              Object {
                "strokeDasharray": "249.75661596038856 471.23889803846896",
                "strokeWidth": 50,
              }
            }
            transform="rotate(-90 100,100)"
          />
          <text
<<<<<<< HEAD
            className="donutchartpercentage-text"
=======
            className="DonutChartPercentage-text"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
            style={
              Object {
                "textAnchor": "middle",
              }
            }
            x={100}
            y={100}
          >
            <tspan
<<<<<<< HEAD
              className="donutchartpercentage-text-val"
            >
              53
            </tspan>
            <tspan
              className="donutchartpercentage-text-percent"
            >
              %
            </tspan>
            <tspan
              className="donutchartpercentage-text-label"
=======
              className="DonutChartPercentage-text-val"
            >
              53
            </tspan>
            <tspan>
              %
            </tspan>
            <tspan
              className="DonutChartPercentage-text-label"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              x={100}
              y={110}
            >
              Completed
            </tspan>
          </text>
        </svg>,
        <Legend
          series={
            Array [
              Object {
                "color": "black",
                "description": "this chart shows a percentage of a value to a target",
                "title": "percentage chart",
              },
            ]
          }
        />,
      ],
      "className": "",
<<<<<<< HEAD
=======
      "description": "",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
      "title": "percentage chart",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <circle
<<<<<<< HEAD
              className="donutchartpercentage-track"
=======
              className="DonutChartPercentage-track"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              cx={100}
              cy={100}
              r={75}
              style={
                Object {
                  "strokeWidth": 50,
                }
              }
              transform="rotate(-90 100,100)"
            />,
            <circle
<<<<<<< HEAD
              className="donutchartpercentage-indicator"
=======
              className="DonutChartPercentage-indicator"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              cx={100}
              cy={100}
              r={75}
              style={
                Object {
                  "strokeDasharray": "249.75661596038856 471.23889803846896",
                  "strokeWidth": 50,
                }
              }
              transform="rotate(-90 100,100)"
            />,
            <text
<<<<<<< HEAD
              className="donutchartpercentage-text"
=======
              className="DonutChartPercentage-text"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              style={
                Object {
                  "textAnchor": "middle",
                }
              }
              x={100}
              y={100}
            >
              <tspan
<<<<<<< HEAD
                className="donutchartpercentage-text-val"
              >
                53
              </tspan>
              <tspan
                className="donutchartpercentage-text-percent"
              >
                %
              </tspan>
              <tspan
                className="donutchartpercentage-text-label"
=======
                className="DonutChartPercentage-text-val"
              >
                53
              </tspan>
              <tspan>
                %
              </tspan>
              <tspan
                className="DonutChartPercentage-text-label"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                x={100}
                y={110}
              >
                Completed
              </tspan>
            </text>,
          ],
<<<<<<< HEAD
          "className": "donutchartpercentage",
=======
          "className": "DonutChartPercentage",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
          "height": 200,
          "width": 200,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
<<<<<<< HEAD
              "className": "donutchartpercentage-track",
=======
              "className": "DonutChartPercentage-track",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              "cx": 100,
              "cy": 100,
              "r": 75,
              "style": Object {
                "strokeWidth": 50,
              },
              "transform": "rotate(-90 100,100)",
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
<<<<<<< HEAD
              "className": "donutchartpercentage-indicator",
=======
              "className": "DonutChartPercentage-indicator",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              "cx": 100,
              "cy": 100,
              "r": 75,
              "style": Object {
                "strokeDasharray": "249.75661596038856 471.23889803846896",
                "strokeWidth": 50,
              },
              "transform": "rotate(-90 100,100)",
            },
            "ref": null,
            "rendered": null,
            "type": "circle",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <tspan
<<<<<<< HEAD
                  className="donutchartpercentage-text-val"
                >
                  53
                </tspan>,
                <tspan
                  className="donutchartpercentage-text-percent"
                >
                  %
                </tspan>,
                <tspan
                  className="donutchartpercentage-text-label"
=======
                  className="DonutChartPercentage-text-val"
                >
                  53
                </tspan>,
                <tspan>
                  %
                </tspan>,
                <tspan
                  className="DonutChartPercentage-text-label"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                  x={100}
                  y={110}
                >
                  Completed
                </tspan>,
              ],
<<<<<<< HEAD
              "className": "donutchartpercentage-text",
=======
              "className": "DonutChartPercentage-text",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              "style": Object {
                "textAnchor": "middle",
              },
              "x": 100,
              "y": 100,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 53,
<<<<<<< HEAD
                  "className": "donutchartpercentage-text-val",
=======
                  "className": "DonutChartPercentage-text-val",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                },
                "ref": null,
                "rendered": 53,
                "type": "tspan",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "%",
<<<<<<< HEAD
                  "className": "donutchartpercentage-text-percent",
=======
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                },
                "ref": null,
                "rendered": "%",
                "type": "tspan",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Completed",
<<<<<<< HEAD
                  "className": "donutchartpercentage-text-label",
=======
                  "className": "DonutChartPercentage-text-label",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                  "x": 100,
                  "y": 110,
                },
                "ref": null,
                "rendered": "Completed",
                "type": "tspan",
              },
            ],
            "type": "text",
          },
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "series": Array [
            Object {
              "color": "black",
              "description": "this chart shows a percentage of a value to a target",
              "title": "percentage chart",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <svg
<<<<<<< HEAD
            className="donutchartpercentage"
=======
            className="DonutChartPercentage"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
            height={200}
            width={200}
          >
            <circle
<<<<<<< HEAD
              className="donutchartpercentage-track"
=======
              className="DonutChartPercentage-track"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              cx={100}
              cy={100}
              r={75}
              style={
                Object {
                  "strokeWidth": 50,
                }
              }
              transform="rotate(-90 100,100)"
            />
            <circle
<<<<<<< HEAD
              className="donutchartpercentage-indicator"
=======
              className="DonutChartPercentage-indicator"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              cx={100}
              cy={100}
              r={75}
              style={
                Object {
                  "strokeDasharray": "249.75661596038856 471.23889803846896",
                  "strokeWidth": 50,
                }
              }
              transform="rotate(-90 100,100)"
            />
            <text
<<<<<<< HEAD
              className="donutchartpercentage-text"
=======
              className="DonutChartPercentage-text"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
              style={
                Object {
                  "textAnchor": "middle",
                }
              }
              x={100}
              y={100}
            >
              <tspan
<<<<<<< HEAD
                className="donutchartpercentage-text-val"
              >
                53
              </tspan>
              <tspan
                className="donutchartpercentage-text-percent"
              >
                %
              </tspan>
              <tspan
                className="donutchartpercentage-text-label"
=======
                className="DonutChartPercentage-text-val"
              >
                53
              </tspan>
              <tspan>
                %
              </tspan>
              <tspan
                className="DonutChartPercentage-text-label"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                x={100}
                y={110}
              >
                Completed
              </tspan>
            </text>
          </svg>,
          <Legend
            series={
              Array [
                Object {
                  "color": "black",
                  "description": "this chart shows a percentage of a value to a target",
                  "title": "percentage chart",
                },
              ]
            }
          />,
        ],
        "className": "",
<<<<<<< HEAD
=======
        "description": "",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
        "title": "percentage chart",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <circle
<<<<<<< HEAD
                className="donutchartpercentage-track"
=======
                className="DonutChartPercentage-track"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                cx={100}
                cy={100}
                r={75}
                style={
                  Object {
                    "strokeWidth": 50,
                  }
                }
                transform="rotate(-90 100,100)"
              />,
              <circle
<<<<<<< HEAD
                className="donutchartpercentage-indicator"
=======
                className="DonutChartPercentage-indicator"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                cx={100}
                cy={100}
                r={75}
                style={
                  Object {
                    "strokeDasharray": "249.75661596038856 471.23889803846896",
                    "strokeWidth": 50,
                  }
                }
                transform="rotate(-90 100,100)"
              />,
              <text
<<<<<<< HEAD
                className="donutchartpercentage-text"
=======
                className="DonutChartPercentage-text"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                style={
                  Object {
                    "textAnchor": "middle",
                  }
                }
                x={100}
                y={100}
              >
                <tspan
<<<<<<< HEAD
                  className="donutchartpercentage-text-val"
                >
                  53
                </tspan>
                <tspan
                  className="donutchartpercentage-text-percent"
                >
                  %
                </tspan>
                <tspan
                  className="donutchartpercentage-text-label"
=======
                  className="DonutChartPercentage-text-val"
                >
                  53
                </tspan>
                <tspan>
                  %
                </tspan>
                <tspan
                  className="DonutChartPercentage-text-label"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                  x={100}
                  y={110}
                >
                  Completed
                </tspan>
              </text>,
            ],
<<<<<<< HEAD
            "className": "donutchartpercentage",
=======
            "className": "DonutChartPercentage",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
            "height": 200,
            "width": 200,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
<<<<<<< HEAD
                "className": "donutchartpercentage-track",
=======
                "className": "DonutChartPercentage-track",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                "cx": 100,
                "cy": 100,
                "r": 75,
                "style": Object {
                  "strokeWidth": 50,
                },
                "transform": "rotate(-90 100,100)",
              },
              "ref": null,
              "rendered": null,
              "type": "circle",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
<<<<<<< HEAD
                "className": "donutchartpercentage-indicator",
=======
                "className": "DonutChartPercentage-indicator",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                "cx": 100,
                "cy": 100,
                "r": 75,
                "style": Object {
                  "strokeDasharray": "249.75661596038856 471.23889803846896",
                  "strokeWidth": 50,
                },
                "transform": "rotate(-90 100,100)",
              },
              "ref": null,
              "rendered": null,
              "type": "circle",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <tspan
<<<<<<< HEAD
                    className="donutchartpercentage-text-val"
                  >
                    53
                  </tspan>,
                  <tspan
                    className="donutchartpercentage-text-percent"
                  >
                    %
                  </tspan>,
                  <tspan
                    className="donutchartpercentage-text-label"
=======
                    className="DonutChartPercentage-text-val"
                  >
                    53
                  </tspan>,
                  <tspan>
                    %
                  </tspan>,
                  <tspan
                    className="DonutChartPercentage-text-label"
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                    x={100}
                    y={110}
                  >
                    Completed
                  </tspan>,
                ],
<<<<<<< HEAD
                "className": "donutchartpercentage-text",
=======
                "className": "DonutChartPercentage-text",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                "style": Object {
                  "textAnchor": "middle",
                },
                "x": 100,
                "y": 100,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": 53,
<<<<<<< HEAD
                    "className": "donutchartpercentage-text-val",
=======
                    "className": "DonutChartPercentage-text-val",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                  },
                  "ref": null,
                  "rendered": 53,
                  "type": "tspan",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "%",
<<<<<<< HEAD
                    "className": "donutchartpercentage-text-percent",
=======
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                  },
                  "ref": null,
                  "rendered": "%",
                  "type": "tspan",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Completed",
<<<<<<< HEAD
                    "className": "donutchartpercentage-text-label",
=======
                    "className": "DonutChartPercentage-text-label",
>>>>>>> 43833c2510545533448b3592e9bdf129403948b3
                    "x": 100,
                    "y": 110,
                  },
                  "ref": null,
                  "rendered": "Completed",
                  "type": "tspan",
                },
              ],
              "type": "text",
            },
          ],
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "series": Array [
              Object {
                "color": "black",
                "description": "this chart shows a percentage of a value to a target",
                "title": "percentage chart",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
